src/main/java/pl/isa/biblioteka/Menu.java

```diff
@@ -56,51 +64,47 @@ public void librarianMenu() {
                     }
                 } else {
                     System.out.println("Wybierz poprawny numer menu");
-                    break;
+                    continue;
                 }
-            } else {
-                System.out.println("Wybierz poprawny numer menu");
-                break;
             }
+        } catch (Exception e) {
+            System.out.println("Wprowadziłeś niepoprawny znak");
+            librarianMenu();
```

Comment:
Podobnie jak wyżej, rekurencja nie jest tu dobrym rozwiązaniem, trzeba by to wszystko w pętli po prostu umieścić.